classlib/src/main/java/org/teavm/classlib/java/util/TObjects.java

```diff
@@ -158,14 +158,14 @@ public static long checkFromToIndex(long fromIndex, long toIndex, long length) {
     }
 
     public static int checkFromIndexSize(int fromIndex, int size, int length) {
-        if (fromIndex < 0 || size < 0 || fromIndex + size > length) {
+        if (fromIndex < 0 || size < 0 || size > length - fromIndex) {
```

Comment:
What was the reason for this change?

Reply:
Corner cases like length = Integer.MAX_VALUE, size = Integer.MAX_VALUE, fromIndex = 2.